cluster-autoscaler/simulator/clustersnapshot/store/delta.go

```diff
@@ -428,24 +455,69 @@ func (snapshot *DeltaSnapshotStore) AddSchedulerNodeInfo(nodeInfo *schedulerfram
 	return nil
 }
 
+// setClusterStatePodsSequential sets the pods in cluster state in a sequential way.
+func (snapshot *DeltaSnapshotStore) setClusterStatePodsSequential(nodeInfos []*schedulerframework.NodeInfo, nodeNameToIdx map[string]int, scheduledPods []*apiv1.Pod) error {
```

Comment:
Do we actually need a separate sequential implementation? It behaves the same as `...Parallelized` one with `parallelism == 1`, right?

Reply:
I decided to leave `parallelism == 1` with previous implementation to make the change potentially revertible (see the discussion above). Sequential implementation could be eventually removed and use `workqueue.ParallelizedUntil` when we are sure the behavior is correct.

Reply:
Sure, it is just that code cleanups are easy to forget. Can you at least leave a TODO here with your github handle to follow up on this?